src/src.pro

```diff
@@ -1,6 +1,12 @@
-TARGET = qofonoext
+equals(QT_MAJOR_VERSION, 6) {
+    QOFONOEXT_SUFFIX = 6
+} else {
+    QOFONOEXT_SUFFIX =
+}
+
+TARGET = qofonoext$${QOFONOEXT_SUFFIX}
```

Comment:
Especially this one. Who knows what's linking against it. I find it highly unlikely that both Qt5  and Qt6 libs would have to coexist on the same system.